NetworkingSampleApp/NetworkingSampleApp/API/Routers/SampleUploadRouter.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  SampleUploadRouter.swift
+//  NetworkingSampleApp
+//
+//  Created by Tony Ngo on 12.06.2023.
+//
+
+import Foundation
+import Networking
+import UniformTypeIdentifiers
+
+enum SampleUploadRouter: Requestable {
+    case image
+    case file(URL)
+
+    var baseURL: URL {
+        fatalError("Provide your API base URL for upload")
+    }
+
+    var headers: [String: String]? {
+        switch self {
+        case .image:
+            return ["Content-Type": "image/png"]
+        case let .file(url):
+            return ["Content-Type": url.mimeType]
+        }
+    }
+
+    var path: String {
+        fatalError("Provide your API endpoint path for upload")
+    }
+
+    var method: HTTPMethod {
+        .post
+    }
+}
+
+private extension URL {
+    var mimeType: String {
+        UTType(filenameExtension: pathExtension)?.preferredMIMEType ?? "application/octet-stream"
```

Comment:
As a sample, it does not matter as much, IMO. In the actual use case, it would be preferred to have the mime type provided by the caller — something like `case file(URL, mimeType: String)`.